cs251tk/toolkit/process_student.py

```diff
@@ -25,22 +25,22 @@ def process_student(
 ):
     if clean:
         remove(student)
-    logging.debug("Cloning    {}'s repository".format(student))
+    logging.debug("Cloning {}'s repository".format(student))
     clone_student(student, baseurl=stogit_url)
 
     try:
-        logging.debug("Stashing   {}'s repository".format(student))
+        logging.debug("Stashing {}'s repository".format(student))
         stash(student, no_update=no_update)
 
-        logging.debug("Pulling    {}'s repository".format(student))
+        logging.debug("Pulling {}'s repository".format(student))
         pull(student, no_update=no_update)
-
-        logging.debug("Checking out commits in {}'s repository before {}".format(student, date))
+        if date:
+            logging.debug("Checking out commits in {}'s repository before {}".format(student, date))
```

Comment:
This conditional makes me think, shouldn't we just put these logs into the functions themselves?

Like, log `pulling repository` from within the `pull` function, instead of the calling function.

Reply:
That makes sense. When I made them originally, I was using the if debug: check which could only be used in the calling function. Because the debug check is basically global, that should work.